spring-boot-autoconfigure/src/test/java/org/axonframework/springboot/autoconfig/InfraConfigurationTest.java

```diff
@@ -0,0 +1,136 @@
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.jmx.support.RegistrationPolicy;
+import org.springframework.test.context.ContextConfiguration;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.*;
+
+
+/**
+ * Test class validating the behavior of the {@link InfraConfiguration}.
+ *
+ * @author Steven van Beelen
+ */
+class InfraConfigurationTest {
+
+    private ApplicationContextRunner testApplicationContext;
+
+    @BeforeEach
+    void setUp() {
+        testApplicationContext = new ApplicationContextRunner().withUserConfiguration(DefaultContext.class)
+                                                               .withPropertyValues("axon.axonserver.enabled:false");
+    }
+
+    @Test
+    void infraConfigurationConstructsSpringConfigurerAndSpringAxonConfiguration() {
+        testApplicationContext.run(context -> {
+            assertThat(context).hasSingleBean(SpringConfigurer.class);
+            assertThat(context).getBean("springAxonConfigurer")
+                               .isInstanceOf(SpringConfigurer.class);
+            assertThat(context).hasSingleBean(SpringAxonConfiguration.class);
+            // The SpringAxonConfiguration is a factory for the DefaultConfigurer.ConfigurationImpl that is private.
+            // Hence, we perform a non-null check.
+            assertThat(context).getBean("springAxonConfiguration")
+                               .isNotNull();
+        });
+    }
+
+    @Test
+    void infraConfigurationConstructsLookUpBeans() {
+        testApplicationContext.run(context -> {
+            assertThat(context).hasSingleBean(MessageHandlerLookup.class);
+            assertThat(context).hasSingleBean(SpringAggregateLookup.class);
+            assertThat(context).hasSingleBean(SpringSagaLookup.class);
+        });
+    }
+
+    @Test
+    void multipleEventUpcasterChainBeansAreCombinedInSingleEventUpcasterChainThroughConfiguration() {
+        //noinspection unchecked
+        Stream<IntermediateEventRepresentation> mockStream = mock(Stream.class);
+
+        testApplicationContext.withUserConfiguration(UpcasterContext.class).run(context -> {
+            assertThat(context).getBean("firstUpcasterChain")
+                               .isInstanceOf(EventUpcasterChain.class);
+            assertThat(context).getBean("secondUpcasterChain")
+                               .isInstanceOf(EventUpcasterChain.class);
+
+            EventUpcasterChain result = context.getBean("springAxonConfiguration", Configuration.class)
+                                               .upcasterChain();
+            assertThat(result).isNotNull();
+
+            result.upcast(mockStream);
+
+            InOrder upcasterOrder = inOrder(mockStream);
```

Comment:
I was very confused by the verifications on `map` and `sorted` here. Can we add a comment explaining that the verifications mean the upcasters have been called because we made them this way on line 122 and 131?

Reply:
Fair one, will do.